miden-lib/asm/miden/kernels/tx/account.masm

```diff
@@ -433,15 +433,25 @@ export.authenticate_procedure.1
     # => [PROC_ROOT, index, CODE_ROOT]
 
     # push the depth of the code Merkle tree onto the stack
-    push.ACCOUNT_CODE_TREE_DEPTH movdn.4
+    # push.ACCOUNT_CODE_TREE_DEPTH movdn.4
     # => [PROC_ROOT, depth, index, CODE_ROOT]
 
     # verify the procedure exists in the account code Merkle tree
-    mtree_verify
+    # mtree_verify
     # => [PROC_ROOT, depth, index, CODE_ROOT]
 
+    # verify that the procedure exists in the account code proc map
+    push.0 eq assert.err=111
+    push.0 assert.err=666
+    # push.0 assert.err=666 dropw adv.push_mapvaln dupw padw adv_loadw eq assert
```

Comment:
Here I have been trying to replace the MT validation to the of the proc_root by asserting that it exists in the advice provider under the correct root. 

For now it has been failing consistently `adv.push_mapvaln dupw padw adv_loadw eq assert` has been returning me:
- this error:
```shell
const MASTS: [&str; 11] = [
    "0x7cb6a9199c2e20758777fb0c55c5960b47696e309a5a90702d4e354b6ecb4389",
    "0x5871dc217fb6e0a8f2b6842db687cb839b05142200de313d5cb665df9253b21d",
    "0xa84b2ff60dbbc6f7877c484bb7ce6f982fa52cd4ec3254c91ebf016eef8fe3ca",
    "0x25a209ab99a801f915bba1e024d55e3c471b123b900037ca967b186b923b9ba3",
    "0x3346321ad53f070e7316baa0c1424c82e064db8302c0027837a1128277491e79",
    "0x2df2ff40ccf245a5030d4e182d12cef40be917ab35026feace19203fcbcc505c",
    "0xfaff73edcf2e8d4264baa3f0ca2bb742923372e1604e28ffda70284ec9df3ec9",
    "0x491e41c5f7b56a2d639e74d2bcea3b65d63a9a3791b448fe368045ca7a57f85b",
    "0x8bd06a4689b30dbdeb4146e3c03a099566685235f2515df047d45fed2413784c",
    "0xff06b90f849c4b262cbfbea67042c4ea017ea0e9c558848a951d44b23370bec5",
    "0x8ef0092134469a1330e3c468f57c7f085ce611645d09cc7516c786fefc71d794",
];
```
or this error: 
```shell
AdviceMapKeyNotFound([6682005582809191751, 7989119676173177425, 12980942404489713001, 255])
```
Or similar.

Reply:
We may need to discuss the logic of what you are trying to do here, because I'm not fully following it.

In general, we need to update several places to make this work. Specifically:
1. In the prologue, we need to "unhash" the procedure commitment into some well-known memory location (this would require updating both [prologue.masm](https://github.com/0xPolygonMiden/miden-base/blob/5a672016aa315c571a20cb3086cfe53aab206423/miden-lib/asm/miden/kernels/tx/prologue.masm) and [memory.masm](https://github.com/0xPolygonMiden/miden-base/blob/5a672016aa315c571a20cb3086cfe53aab206423/miden-lib/asm/miden/kernels/tx/memory.masm)).
2. Then this procedure could be pretty simple - probably something like this:

```
#! Stack: [PROC_ROOT]
#! Output: [storage_offset]
export.authenticate_procedure
    # load the index of the procedure onto the advice stack and move it to the operand stack
    emit.ACCOUNT_PUSH_PROCEDURE_INDEX_EVENT adv_push.1
    # => [index, PROC_ROOT]

    # get the procedure info (root, storage_offset) of the account procedure stored at the
    # index we got above
    exec.get_account_procedure_info
    # => [STORED_PROC_ROOT, storage_offset, PROC_ROOT]

    # make sure stored proc root is the same as we got with the parameter
    movup.4 movdn.8 assert_eqw
    # => [storage_offset]
end
```

The above assumes that we have `get_account_procedure_info` procedure which reads the procedure info from kernel memory (from the place we wrote it into in step 1 above) and puts it onto the stack.